iroha/src/asset.rs

```diff
@@ -21,68 +24,218 @@ impl Asset {
     /// This method should not be used to create an `Asset` to work with as a part of the Iroha
     /// State.
     pub fn new(id: Id) -> Self {
-        Asset { id, quantity: 0 }
+        Asset { id }
     }
+}
+
+/// All possible variants of `Asset` entity's components.
+#[derive(Clone, Debug, Encode, Decode, Eq, Ord, PartialEq, PartialOrd)]
+pub enum Component {
```

Comment:
There are only 3 fields that an asset can have, name, (big)qty, and a kv store. Besides big quantity and quantity, an asset can have all at once. I don't see the need for an enum as these can just be optional fields.

Reply:
Ok